src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java

```diff
@@ -529,4 +553,354 @@ public static void main(String[] args) {
     }
     System.exit(1);
   }
+
+  /**
+   * Fetches all the translations necessary to construct the sight and light help text
+   *
+   * @return Map of keys to translations
+   */
+  private Map<String, String> createSightLightHelpTextMap() {
+    Map<String, String> parameters = new HashMap<>();
+    /* cell formatting string */
+    parameters.put("alignCellCenter", " align=center");
+    /* Useful words and phrases */
+    parameters.put("phraseMultipleEntriesAllowed", I18N.getText("phrase.multipleEntriesAllowed"));
+    parameters.put("mapVisionDistance", I18N.getText("sight.default.distance"));
+
+    /* Build list of useful words and phrases */
+    List<String> helpKeys = I18N.getMatchingKeys(Pattern.compile("^word."));
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^option.type.")));
+    /* Shape names */
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^shape.type.name.")));
+    /* example text */
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^sight.example.")));
+    helpKeys.addAll(I18N.getMatchingKeys(Pattern.compile("^light.example.")));
+
+    /* Generate parameter map from list */
+    for (String key : helpKeys) {
+      parameters.putIfAbsent(
+          uncapitalize(capitalize(key, '.').replace(".", "")), I18N.getText(key));
+    }
+
+    /* Parameterised strings - need to be done individually */
+    parameters.put(
+        "wikiLinkReferral",
+        I18N.getText(
+            "sightLight.wikiLinkReferral",
+            "<i>wiki.rptools.info/index.php/Introduction_to_Lights_and_Sights</i>"));
```

Comment:
This format < i > sticks out like a sore thumb. Is it really necessary here?

Reply:
I can't format it as a link (preferred) because it won't actually open in a browser (or at least it didn't when I tried). I wanted it to stand out so people know what to copy-paste. I can't hard-code the colour because that may not be visible in some themes and linking to theme.css didn't work for utilising theme-sensitive styling like accent colour.
That left me with bold, italic, and underline as means of differentiation. Underline implies clickable which is inappropriate and bold fails my aesthetic text for the second part of a sentence. That left me with italic, the best of an admittedly stagnant pool of options.

Reply:
Does it have to be formatted at all? This is the only place I saw, where formatting is part of the "value". The other sections have the formatting in the "container". E.g., here it could be "...<i>{sightLight.wikiLinkReferral}</i>..."

Reply:
In order to help keep the translation text context I'm passing it as a parameter.
So the translator sees something like "referral.wikilink = find more in the wiki online at {0}."
Because it is being injected into the sentence it needs to carry its own formatting if I want to keep the tags out of the translatable text.
Formatting is not necessary, but I want it, so I coded it to be kept out of the way of everyone except developers.

Reply:
I see, that part is never to be translated anyway. The chances are less than slim, that this would ever be referenced as wiki.rptools.info/index.php/es/introduccion_a_luces_y_vistas...